### DISCUSSION TITLE
request query_param parse error when + in value

### DISCUSSION THREAD

**Discussion by cst4049** [fastapi/fastapi#6956] [Questions] [answered]
### First check
http://localhost:8000/test?a=b+c&q=search
query_param like this
{"a": "b c","q":"search"}
this parse might be a bug

![image](http://user-images.githubusercontent.com/16145670/104309801-aec4c900-550d-11eb-8194-4b4213167986.png)

![image](http://user-images.githubusercontent.com/16145670/104309725-9359be00-550d-11eb-966c-9d7fc12de460.png)

### **right click can open the picture show** 

request.query_param.get('a')  got 'b c' instead of 'b+c'

when query_param.get('key') it '+' was replaced by ' '

**Comment by ycd** [comment]
Can you provide a reproducible example? How your endpoint looks like etc.

**Comment by cst4049** [comment]
url localhost:8000/test?a=b+c

code like this
~~~
@app.route('/test', methods=['get'])
def heartbeat(request: Request):
    params = request.query_params
    params.get('a')  # 'b c' instead 'b+c'
~~~

**Comment by Mause** [comment, marked as answer]
That is correct, + means space in URLs - if you want a literal + you'll need to urlencode it

**Comment by cst4049** [comment]
yes just encoding it

```
urllib parse.py

        if len(nv[1]) or keep_blank_values:
            name = nv[0].replace('+', ' ')
            name = unquote(name, encoding=encoding, errors=errors)
            name = _coerce_result(name)
            value = nv[1].replace('+', ' ')    # this replace '+' ==> ' '
            value = unquote(value, encoding=encoding, errors=errors)
            value = _coerce_result(value)
            r.append((name, value))
```

thanks for help

**Comment by tiangolo** [comment]
Thanks for the help here @ycd and @Mause ! 👏  🙇 

Thanks for reporting back and closing the issue @cst4049  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
